huaweicloud/services/dms/resource_huaweicloud_dms_kafka_smart_connect_task.go

```diff
@@ -0,0 +1,378 @@
+	getKafkaSmartConnectTaskPath = strings.ReplaceAll(getKafkaSmartConnectTaskPath, "{project_id}",
+		getKafkaSmartConnectTaskClient.ProjectID)
+	getKafkaSmartConnectTaskPath = strings.ReplaceAll(getKafkaSmartConnectTaskPath, "{connector_id}", connectorID)
+	getKafkaSmartConnectTaskPath = strings.ReplaceAll(getKafkaSmartConnectTaskPath, "{task_id}", d.Id())
+
+	getKafkaSmartConnectTaskOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	getKafkaSmartConnectTaskResp, err := getKafkaSmartConnectTaskClient.Request("GET", getKafkaSmartConnectTaskPath,
+		&getKafkaSmartConnectTaskOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DMS kafka smart connect task")
+	}
+
+	getKafkaSmartConnectTaskRespBody, err := utils.FlattenResponse(getKafkaSmartConnectTaskResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("connector_id", connectorID),
+		d.Set("task_name", utils.PathSearch("task_name", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("destination_type", utils.PathSearch("destination_type", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("created_at", utils.PathSearch("create_time", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("status", utils.PathSearch("status", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("topics", utils.PathSearch("topics", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("obs_destination_descriptor", flattenObsDestinationDescriptor(getKafkaSmartConnectTaskRespBody)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenObsDestinationDescriptor(getKafkaSmartConnectTaskRespBody interface{}) []map[string]interface{} {
+	if getKafkaSmartConnectTaskRespBody == nil {
+		return nil
+	}
+
+	obsDestinationDescriptorStruct := utils.PathSearch("obs_destination_descriptor", getKafkaSmartConnectTaskRespBody, nil)
+	if obsDestinationDescriptorStruct == nil {
+		return nil
+	}
+
+	obsDestinationDescriptorList := make([]map[string]interface{}, 1)
+	params := make(map[string]interface{})
+	params["topics"] = utils.PathSearch("topics", obsDestinationDescriptorStruct, nil)
+	params["topics_regex"] = utils.PathSearch("topics_regex", obsDestinationDescriptorStruct, nil)
+	params["consumer_strategy"] = utils.PathSearch("consumer_strategy", obsDestinationDescriptorStruct, nil)
+	params["destination_file_type"] = utils.PathSearch("destination_file_type", obsDestinationDescriptorStruct, nil)
+	params["obs_bucket_name"] = utils.PathSearch("obs_bucket_name", obsDestinationDescriptorStruct, nil)
+	params["obs_path"] = utils.PathSearch("obs_path", obsDestinationDescriptorStruct, nil)
+	params["partition_format"] = utils.PathSearch("partition_format", obsDestinationDescriptorStruct, nil)
+	params["record_delimiter"] = utils.PathSearch("record_delimiter", obsDestinationDescriptorStruct, nil)
+	params["deliver_time_interval"] = utils.PathSearch("deliver_time_interval", obsDestinationDescriptorStruct, nil)
```

Comment:
suggest to write like this
```suggestion
	params := map[string]interface{}{
		"topics": utils.PathSearch("topics", obsDestinationDescriptorStruct, nil),
		"topics_regex": utils.PathSearch("topics_regex", obsDestinationDescriptorStruct, nil),
		...
	}
```

Reply:
done